test/integration/auto-export/pages/[post]/[cmnt].js

```diff
@@ -0,0 +1,21 @@
+import { useRouter } from 'next/router'
+
+if (typeof window !== 'undefined') {
+  window.caughtWarns = []
+  const origWarn = window.console.warn
+  const origError = window.console.error
+  window.console.warn = function (...args) {
+    window.caughtWarns.push(1)
+    origWarn(...args)
+  }
+  window.console.error = function (...args) {
+    window.caughtWarns.push(1)
+    origError(...args)
+  }
+}
+
+export default () => {
+  const { query, asPath } = useRouter()
+  const { cmnt } = query
+  return <p>{cmnt || asPath}</p>
```

Comment:
This test seems pretty hard to understand, can't you add 2 fields instead?

Reply:
This is testing that the hydration warning shows in development when `asPath` is used on an auto-prerendered dynamic-route. I simplified it down to `<p>{useRouter().asPath}</p>` since that's the main thing being tested.

We could also take it a step further and prevent the hydration from breaking by having `asPath` match the auto-prerendered value during CSR and update it when we parse the dynamic values on mount